client.go

```diff
@@ -68,11 +68,10 @@ func (c Client) Send(filename string, mode string) (io.ReaderFrom, error) {
 		s.opts["blksize"] = strconv.Itoa(c.blksize)
 	}
 	n := packRQ(s.send, opWRQ, filename, mode, s.opts)
-	addr, err := s.sendWithRetry(n)
+	err = s.sendWithRetry(n)
 	if err != nil {
 		return nil, err
 	}
-	s.addr = addr
```

Comment:
Yes, that looks like the reason send does not work in tests? That is the side effect of the fact `sender` is re-used by `Server` and `Client` and for client we can't use `net.DialUDP` because we will receive packets from the different port, not 69. Tricky.